storage/object/api-cli/combining-iam-and-object-storage.mdx

```diff
@@ -0,0 +1,319 @@
+
+Bucket policies behave like objects: they can be uploaded into buckets. Once you upload, or "put", a bucket policy into a bucket, they take immediate effect and will from then on define who can access and perform actions on the bucket.
+
+## Defining whether to use IAM or bucket policies
+
+IAM can be used when you do not need to configure fine-grained access to specific buckets, and favor simplicity over granularity. IAM is central, and policy-based: it says where people can perform actions in the entire Scaleway ecosystem. For example, if access to resources at project level is enough for your Object Storage use cases, we advise you to use IAM only.
+
+Bucket policies operate at the bucket level, and enable you to state clearly who can perform different actions on which specific bucket.
+
+If you need to configure more granular access to your Object Storage buckets, you can combine both IAM and bucket policies. For example, if you must restrict access to specific actions you can perform on a bucket, or if you need the bucket to be available only to certain users, you can use a combination of IAM and bucket policies.
+
+## Combining IAM and bucket policies
+
+In order to perform actions on a bucket, a principal (user or application) must be granted access via an IAM policy and/or an existing bucket policy.
+
+For example, if a user wants to list the objects in a bucket, they must ensure that:
+
+    - the IAM policy explicitly allows the user to list Objects (for example, with an `ObjectStorageReadOnly` or a `ObjectStorageObjectRead` permission set specified in the policy)
+    - if there is no existing bucket policy, the action allowed by IAM is enough to grant them list access
+    - if there is a bucket policy, the list action is explicitly allowed in the policy
+
+<Message type="important">
+  There are a couple of important things to keep in mind when using bucket and IAM policies concurrently:
+    - Even if you only want to define the permissions available for bucket policies, you must first define IAM policy permissions. This is a necessary step of the process.
+    - when you create a Scaleway API key with IAM, you are prompted to specify the API key’s [preferred Project for Object Storage](/identity-and-access-management/iam/api-cli/using-api-key-object-storage/). This API key will always use this Project when carrying out Object Storage actions via any API/CLI.
+    - the preferred Project must always be the one of the bucket where you wish to perform an action.
+</Message>
+
+## Examples
+
+In the sections below we provide different examples of how to combine IAM and bucket policies.
+
+### Allowing all actions except DeleteObject
+
+In the example below, we provide the IAM and bucket policy configurations needed to allow an application to perform all actions in a bucket except to delete it.
+
+An IAM policy must be created to grant full access to Object Storage resources in Project 1 to application A.
+
```

Comment:
Here, we say that we are gonna use uuid for all the examples, don't forget to replace them with your own resources'uuid